react/features/welcome/components/VideoSwitch.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+import React, { Component } from 'react';
+import {
+    Switch,
+    TouchableWithoutFeedback,
+    View
+} from 'react-native';
+import { connect } from 'react-redux';
+
+import { translate } from '../../base/i18n';
+import { updateProfile } from '../../base/profile';
+import { Header, Text } from '../../base/react';
+
+import styles, {
+    SWITCH_THUMB_COLOR,
+    SWITCH_UNDER_COLOR
+} from './styles';
+
+type Props = {
+
+    /**
+     * The Redux dispatch functions.
+     */
+    dispatch: Function,
+
+    /**
+     * The i18n translate function.
+     */
+    t: Function,
+
+    /**
+     * The current profile settings from Redux.
+     */
+    _profile: Object
+};
+
+/**
+ * Renders the audio-video switch on the welcome screen.
+ */
+class VideoSwitch extends Component<Props> {
+    /**
+     * Constructor of the component.
+     *
+     * @inheritdoc
+     */
+    constructor(props) {
+        super(props);
+
+        this._onStartAudioOnlyChange = this._onStartAudioOnlyChange.bind(this);
+    }
+
+    /**
+     * Implements React Component's render.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { t, _profile } = this.props;
+        const { textStyle } = Header;
+
+        return (
+            <View style = { styles.audioVideoSwitchContainer }>
+                <TouchableWithoutFeedback
+                    onPress = {
+                        this._onStartAudioOnlyChangeFn(false)
+                    } >
+                    <Text style = { textStyle } >
+                        { t('welcomepage.audioVideoSwitch.video') }
+                    </Text>
+                </TouchableWithoutFeedback>
+                <Switch
+                    onTintColor = { SWITCH_UNDER_COLOR }
+                    onValueChange = { this._onStartAudioOnlyChange }
+                    style = { styles.audioVideoSwitch }
+                    thumbTintColor = { SWITCH_THUMB_COLOR }
+                    value = { _profile.startAudioOnly } />
+                <TouchableWithoutFeedback
+                    onPress = {
+                        this._onStartAudioOnlyChangeFn(true)
```

Comment:
It's kind of pointless, suboptimal, sending the wrong impression to create a new function on each render, wouldn't you say? This should've been a function created once in the contractor of VideoSwitch.